lua/sp.c

```diff
@@ -4666,6 +4668,57 @@ static int db_error(lua_State *lua)
     return 1;
 }
 
+static int db_guid(Lua lua)
+{
+    int nargs = lua_gettop(lua);
+
+    if (nargs > 2)
+        return luaL_error(lua, "Function may take only one optional guid as string argument, however %d arguments passed in", (nargs - 1));
+
+    uuid_t guid;
+    if (nargs == 1)
+        uuid_generate(guid);
+    else {
+        const char *z = luabb_tostring(lua, 2);
+        if(uuid_parse(z, guid) != 0)
+            return luaL_error(lua, "Can not convert string %s to guid", z);
+    }
+
+    uint8_t *b = malloc(sizeof(guid));
+    memcpy(b, guid, sizeof(guid));
+
+    //sqlite3_result_blob(context, (char*)guid, sizeof(uuid_t)
+    //luabb_pushdatetime(lua, &datetime);
+    blob_t x = {.data = b, .length = sizeof(uuid_t)};
+    luabb_pushblob_dl(lua, &x);
+    return 1;
+}
+
+#define GUID_STR_LENGTH 37
+
+static int db_guid_str(Lua lua)
+{
+    int nargs = lua_gettop(lua);
+
+    if (nargs > 2)
+        return luaL_error(lua, "Function may take only one optional guid as blob argument, however %d arguments passed in", (nargs - 1));
```

Comment:
Minor nitpick - I prefer more idiomatic  error checking and error messages. Can you use luaL_check*, luaL_opt*, luaL_arg* for these?

Reply:
Sure, I can use some of those--I only copied pattern in other functions here such as db_table_insert().